aten/src/ATen/BatchedTensorImpl.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <bitset>
+
+#include <ATen/ArrayRef.h>
+#include <ATen/SmallVector.h>
+#include <ATen/Tensor.h>
+
+namespace at {
+
+// We assume this in a few other places in the codebase,
+// but there isn't a centralized definition.
+constexpr int64_t kVmapMaxTensorDims = 64;
```

Comment:
Why do you choose this constraint / number in particular?

Reply:
If you create a tensor with more than 64 dims, a bunch of things in PyTorch will fail. Off of the top of my head:
- TensorIterator reduction operations will fail because they assume the input can have no more than 64 dims. (https://github.com/pytorch/pytorch/issues/23159)